build-packages.sh

```diff
@@ -11,8 +11,36 @@ working_tree_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 build_packages_log=$working_tree_root/build-packages.log
 binclashlog=$working_tree_root/binclash.log
 binclashloggerdll=$working_tree_root/Tools/Microsoft.DotNet.Build.Tasks.dll
+RuntimeOS=ubuntu
 
-options="/m /nologo /v:minimal /clp:Summary /flp:v=detailed;Append;LogFile=$build_packages_log /l:BinClashLogger,$binclashloggerdll;LogFile=$binclashlog"
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case $OSName in
```

Comment:
Somewhat orthogonal but we seem to be duplicating this logic in various .sh files we should probably figure out a way to share this logic. @mellinoe made a suggestion to be on friday which I think kind of makes some since for the dev workflow logic. @maririos we should consider adding an msbuild.cmd/sh script that captures a lot of this common logic (i.e. OSGroup selection, common logging, init-tools, etc) and have all these other wrappers simply call that msbuild script with the extra arguments unique to that script. That will allow us to encapsulate a lot of this logic and also provide a way for us to scope the need for having to add msbuild to the path.